From 9a7d2cb06ea9c1ce37f41af9af3222f2d939ec34 Mon Sep 17 00:00:00 2001
From: Teemu Katajisto <ext-teemu.katajisto@nokia.com>
Date: Wed, 4 Nov 2009 12:46:09 +0200
Subject: [PATCH] camerabin: change queue place in videobin (Maemo-specific)

Performance increase when queue element is placed between video
encoder and muxer.
---
 gst/camerabin/camerabinvideo.c |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/gst/camerabin/camerabinvideo.c b/gst/camerabin/camerabinvideo.c
index afd36b7..3ccf22d 100644
--- a/gst/camerabin/camerabinvideo.c
+++ b/gst/camerabin/camerabinvideo.c
@@ -556,12 +556,6 @@ gst_camerabin_video_create_elements (GstCameraBinVideo * vid)
   /* Add queue element for video */
   vid->tee_video_srcpad = gst_element_get_request_pad (vid->tee, "src%d");
 
-  vid->video_queue = gst_element_factory_make ("queue", "video-queue");
-  if (!gst_camerabin_add_element (vidbin, vid->video_queue)) {
-    goto error;
-  }
-  g_object_set (vid->video_queue, "silent", TRUE, NULL);
-
   /* Add probe for rewriting video timestamps */
   vid->vid_tee_probe_id = gst_pad_add_buffer_probe (vid->tee_video_srcpad,
       G_CALLBACK (camerabin_video_pad_tee_src0_have_buffer), vid);
@@ -585,6 +579,11 @@ gst_camerabin_video_create_elements (GstCameraBinVideo * vid)
     goto error;
   }
 
+  vid->video_queue = gst_element_factory_make ("queue", "video-queue");
+  if (!gst_camerabin_add_element (vidbin, vid->video_queue)) {
+    goto error;
+  }
+  
   /* Add application set or default muxer element */
   if (vid->app_mux) {
     vid->muxer = vid->app_mux;
